gazelle/config.go

```diff
@@ -0,0 +1,253 @@
+func (c *TypeScriptConfig) SetGenerationEnabled(enabled bool) {
+	c.generationEnabled = enabled
+}
+
+// GenerationEnabled returns whether the extension is enabled or not.
+func (c *TypeScriptConfig) GenerationEnabled() bool {
+	return c.generationEnabled
+}
+
+// Adds a dependency to the list of ignored dependencies for
+// a given package. Adding an ignored dependency to a package also makes it
+// ignored on a subpackage.
+func (c *TypeScriptConfig) AddIgnoredImport(imp string) {
+	c.ignoreDependencies.Add(imp)
+}
+
+// Checks if a dependency is ignored in the given package or
+// in one of the parent packages up to the workspace root.
+func (c *TypeScriptConfig) IsDependencyIgnored(dep string) bool {
+	config := c
+	for config != nil {
+		if config.ignoreDependencies.Contains(dep) {
+			return true
+		}
+		config = config.parent
+	}
+
+	return false
+}
+
+// SetValidateImportStatements sets whether TypeScript import statements should be
+// validated or not. It throws an error if this is set multiple times, i.e. if
+// the directive is specified multiple times in the Bazel workspace.
+func (c *TypeScriptConfig) SetValidateImportStatements(validate bool) {
+	c.validateImportStatements = validate
+}
+
+// ValidateImportStatements returns whether the TypeScript import statements should
+// be validated or not. If this option was not explicitly specified by the user,
+// it defaults to true.
+func (c *TypeScriptConfig) ValidateImportStatements() bool {
+	return c.validateImportStatements
+}
+
+// SetLibraryNamingConvention sets the ts_project target naming convention.
+func (c *TypeScriptConfig) SetLibraryNamingConvention(libraryNamingConvention string) {
+	c.libraryNamingConvention = libraryNamingConvention
+}
+
+// RenderLibraryName returns the ts_project target name by performing all
+// substitutions.
+func (c *TypeScriptConfig) RenderLibraryName(packageName string) string {
+	return strings.ReplaceAll(c.libraryNamingConvention, packageNameNamingConventionSubstitution, packageName)
+}
+
+// SetTestsNamingLibraryConvention sets the ts_project test target naming convention.
+func (c *TypeScriptConfig) SetTestsNamingLibraryConvention(testsNamingConvention string) {
+	c.testsNamingConvention = testsNamingConvention
+}
+
+// RenderTestName returns the ts_project test target name by performing all
+// substitutions.
+func (c *TypeScriptConfig) RenderTestsLibraryName(packageName string) string {
+	return strings.ReplaceAll(c.testsNamingConvention, packageNameNamingConventionSubstitution, packageName)
+}
+
+func (c *TypeScriptConfig) SetSourceFileGlob(srcsFileGlob string) {
+	c.srcsFileGlob = srcsFileGlob
+}
+func (c *TypeScriptConfig) IsSourceFile(filePath string) bool {
+	if !isSourceFileType(filePath) {
+		return false
+	}
+	if c.srcsFileGlob == "" {
+		return true
+	}
+
+	m, e := doublestar.Match(c.srcsFileGlob, filePath)
+
+	if e != nil {
+		fmt.Println("ERROR: ", fmt.Errorf("srcs file glob error %e", e))
```

Comment:
```suggestion
	m, err := doublestar.Match(c.srcsFileGlob, filePath)
	if err != nil {
		fmt.Printf("ERROR: srcs file glob error: %v\n", err)
```

Reply:
Interesting that you had `%e` for error, but it's meant for scientific notation of floating-point numbers.
https://pkg.go.dev/fmt

Reply:
Also, this may beg a bigger question: do you care about the error at all? From the `Match` comment:
```
// Match requires pattern to match all of name, not just a substring.
// The only possible returned error is ErrBadPattern, when pattern
// is malformed.
```

Reply:
> Also, this may beg a bigger question: do you care about the error at all? From the Match comment:

This means the user (developer) set an invalid directive so I think outputting an error and failing is a clear indication to the user. Where simply ignoring the error would add more files then necessary to the BUILD and could easily go unnoticed by the dev?